docs/stdlibs.md

```diff
@@ -0,0 +1,187 @@
+and encoding and decoding of varints.
+
+[...]
+
+var BigEndian bigEndian
+var LittleEndian littleEndian
+type AppendByteOrder interface{ ... }
+type ByteOrder interface{ ... }
+$ gno doc -u -src encoding/binary littleEndian.AppendUint16
+package binary // import "encoding/binary"
+
+func (littleEndian) AppendUint16(b []byte, v uint16) []byte {
+        return append(b,
+                byte(v),
+                byte(v>>8),
+        )
+}
+```
+
+`gno doc` will work automatically when used within the Gno repository or any
+repository which has a `go.mod` dependency on `github.com/gnolang/gno`, which
+can be a simple way to set up your Gno repositories to automatically support
+`gno` commands (aside from `doc`, also `test`, `run`, etc.).
+
+Another alternative is setting your enviornment variable `GNOROOT` to point to
+where you cloned the Gno repository. You can set this in your `~/.profile` file
+to be automatically set up in your console:
+
+```sh
+export GNOROOT=$HOME/gno
+```
+
+## Test standard libraries
+
+There are some additional standard library functions and packages which are
+currently available only in `_test.gno` and `_filetest.gno` files. At the time
+of writing, these are only some additions in the `std` package to support
+changing some values in test functions.
+
+`gno doc` currently doesn't support reading from the test standard libraries,
+though support is planned to be added. For now, you can inspect the directory
+`gnovm/tests/stdlibs`.
+
+## Adding new standard libraries
+
+New standard libraries may be added by simply creating a new directory (whose
+path relative to the `stdlibs` directory will be the import path used in Gno
+programs). Following that, the suggested approach for adding a Go standard
+libary is to copy the original files from the Go source tree, and renaming their
+extensions from `.go` to `.gno`.
+
+> As a small aid, this bash one-liner can be useful to convert all the file
+> extensions:
+>
+> ```sh
+> for i in *.go; do mv $i "$(echo $i | sed 's/\.go$/.gno/')"; done
+> ```
+
+Following that, the suggested approach is to iteratively try running `gno test .`,
+while fixing any errors that may come out of trying to test the package.
+
+Some things to keep in mind:
+
+- Gno doesn't support assembly functions and build tags. Some Go packages may
+  contain assembly versions for different architecture and a `generic.go` file
+  containing the architecture-independent version. The general approach is that
+  of removing everything architecture/os-specific except for the `generic.go` file.
+- Gno doesn't support reflection at the time of writing, which means that for
+  now many packages which rely heavily on reflection have to be delayed or
+  reduced while we figure out the details on how to implement reflection.
+  Aside from the `reflect` package itself, this also translates to very common
+  packages still not available in Gno, such as `fmt` or `encoding/json`.
```

Comment:
- Include the Go version number from which the stdlib was included.
- Comment all changes from the original file with XXX as needed.